packages/vaadin-app-layout/src/vaadin-app-layout.js

```diff
@@ -541,6 +547,10 @@ class AppLayoutElement extends ElementMixin(ThemableMixin(PolymerElement)) {
 
     this._updateDrawerHeight();
 
+    if (this.overlay !== overlay) {
+      this.notifyResize();
```

Comment:
To me it seems like there will be 2 calls for `notifyResize` when window is resized:
- first call caused by the event listener added in `IronResizableBehavior`
- second call from here, after we set the overlay mode.

This is ok, as we need another resize notification when layout overlay mode changes. 
I would suggest to add a test for this by dispatching a fake `resize` event and using `spy` for `notifyResize`.

Reply:
Yes, it might trigger twice in case of a `resize` event, if the resize results in a change of the overlay mode. I wasn't sure if the explicit call above can be removed - if the listener for `resize` triggers before `_updateOverlayMode` then the child might update before the overlay is toggled. That's why I added the call explicitely.

Can you please elaborate what I should assert in the test - that `notifyResize` is called twice if overlay mode changes due to the resize event?

An alternative thought - maybe we can remove the custom `resize` listener and override `notifyResize` to handle the overlay logic instead. Then the flow could be:
- `resize` event
- `IronResizableBehavior` triggers `notifyResize` on app layout
- in `notifyResize` run `_updateOverlayMode`
- and only after that notify children somehow (call `super.notifyResize()` after?)

Reply:
> in notifyResize run _updateOverlayMode
> and only after that notify children somehow (call super.notifyResize() after?)

Sounds good to me 👍   We can indeed update overlay mode and then call `super.notifyResize`.

Reply:
I gave it a try, but I was not able to solve this in a reasonable timeframe. Moving the resize event handling logic into `notifyResize` itself causes several issues. The issue I got stuck with is this scenario:
- `resize` event
- `notifyResize` called
- `_updateOverlayMode` called
- might change `drawerOpened` before `_updateOverlayMode` has finished
- that in turn calls the `drawerOpened` observer and `notifyResize` again
- which runs `_updateOverlayMode` again while the previous call has still not concluded - this should probably never happen and causes tests to fail

Maybe I'm missing something obvious, but it almost seems like you want to separate:
- reacting to window size changes
- calling `notifyResize` explicitely to notify children about size changes

Not sure how to achieve this with iron-resizable alone. Maybe refactoring the component could solve this too, but I'm not comfortable doing this and it would certainly eat up more time.

TLDR: I would suggest to go with the current solution, if there are additional tests required for that I can add them. I'm also up for going through this together with someone more knowledgeable if there is an obvious solution to the problem.

Reply:
@web-padawan Ping.

Reply:
Sorry, I missed this last week. Thank you for the detailed description. I agree that current solution makes sense.
We can consider changing the current approach in future when moving away from `IronResizableBehavior`.